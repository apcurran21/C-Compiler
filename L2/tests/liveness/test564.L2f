(@fib
	1
	%num <- rdi
	%computes_remaining <- %num
	%computes_started <- 0
	%fib_i <- 0
	%fib_im1 <- 0
	:call_label_ciao_1
	rdi <- %computes_started
	mem rsp -8 <- :call_label_ciao_6
	call @encode_and_print 1
	:call_label_ciao_6
	rdi <- %fib_i
	mem rsp -8 <- :call_label_ciao_7
	call @encode_and_print 1
	:call_label_ciao_7
	rdi <- 888888888888888888
	mem rsp -8 <- :call_label_ciao_8
	call @encode_and_print 1
	:call_label_ciao_8
	:call_label_ciao_2
	%done <- %computes_remaining = 0
	cjump %done = 1 :call_label_ciao_5
	:call_label_ciao_3
	%computes_remaining -= 1
	%computes_started += 1
	%past_base_case <- 1 < %computes_started
	cjump %past_base_case = 1 :call_label_ciao_4
	%fib_i += 1
	goto :call_label_ciao_1
	:call_label_ciao_4
	%old_fib_i <- %fib_i
	%fib_i += %fib_im1
	%fib_im1 <- %old_fib_i
	goto :call_label_ciao_1
	:call_label_ciao_5
	return
)
