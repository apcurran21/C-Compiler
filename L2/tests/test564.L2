(@main
  (@main
    0
    :call_label0
    goto :call_label3
    :call_label1
    return
    :call_label2
    %newVar1 <- 0
    %newVar2 <- %arr
    %newVar2 += 24
    %newVar3 <- mem %newVar2 0 
    %newVar5 <- 1
    %newVar4 <- %newVar3
    %newVar4 >>= 1
    %newVar5 <- %newVar5
    %newVar5 *= %newVar4
    %newVar6 <- %i
    %newVar6 *= %newVar5
    %newVar1 <- %newVar1
    %newVar1 += %newVar6
    %newVar1 <- %newVar1
    %newVar1 += 0
    %newVar0 <- %newVar1
    %newVar0 *= 8
    %newVar0 <- %newVar0
    %newVar0 += 32
    %newVar7 <- %arr
    %newVar7 += %newVar0
    %t <- mem %newVar7 0 
    %t <- %t
    %t <<= 1
    %t <- %t
    %t += 1
    rdi <- %t
    call print 1
    %i <- %i
    %i += 1
    %c <- %i < %arrsize
    
    cjump %c = 1 :call_label2
    goto :call_label1
    :call_label3
    %arrsize <- 31
    %arrval <- 3
    %newVar9 <- %arrsize
    %newVar9 >>= 1
    %newVar8 <- %newVar9
    %newVar10 <- %arrval
    %newVar10 >>= 1
    %newVar8 <- %newVar8
    %newVar8 *= %newVar10
    %newVar8 <- %newVar8
    %newVar8 <<= 1
    %newVar8 <- %newVar8
    %newVar8 += 1
    %newVar8 <- %newVar8
    %newVar8 += 6
    rdi <- %newVar8
    rsi <- 1
    call allocate 2
    %arr <- rax
    %newVar11 <- %arr
    %newVar11 += 8
    mem %newVar11 0  <- 5
    %newVar12 <- %arr
    %newVar12 += 16
    mem %newVar12 0  <- %arrsize
    %newVar13 <- %arr
    %newVar13 += 24
    mem %newVar13 0  <- %arrval
    %newVar15 <- 0
    %newVar16 <- %arr
    %newVar16 += 24
    %newVar17 <- mem %newVar16 0 
    %newVar19 <- 1
    %newVar18 <- %newVar17
    %newVar18 >>= 1
    %newVar19 <- %newVar19
    %newVar19 *= %newVar18
    %newVar20 <- 0
    %newVar20 *= %newVar19
    %newVar15 <- %newVar15
    %newVar15 += %newVar20
    %newVar15 <- %newVar15
    %newVar15 += 0
    %newVar14 <- %newVar15
    %newVar14 *= 8
    %newVar14 <- %newVar14
    %newVar14 += 32
    %newVar21 <- %arr
    %newVar21 += %newVar14
    mem %newVar21 0  <- 43
    %newVar23 <- 0
    %newVar24 <- %arr
    %newVar24 += 24
    %newVar25 <- mem %newVar24 0 
    %newVar27 <- 1
    %newVar26 <- %newVar25
    %newVar26 >>= 1
    %newVar27 <- %newVar27
    %newVar27 *= %newVar26
    %newVar28 <- 1
    %newVar28 *= %newVar27
    %newVar23 <- %newVar23
    %newVar23 += %newVar28
    %newVar23 <- %newVar23
    %newVar23 += 0
    %newVar22 <- %newVar23
    %newVar22 *= 8
    %newVar22 <- %newVar22
    %newVar22 += 32
    %newVar29 <- %arr
    %newVar29 += %newVar22
    mem %newVar29 0  <- 73
    %newVar31 <- 0
    %newVar32 <- %arr
    %newVar32 += 24
    %newVar33 <- mem %newVar32 0 
    %newVar35 <- 1
    %newVar34 <- %newVar33
    %newVar34 >>= 1
    %newVar35 <- %newVar35
    %newVar35 *= %newVar34
    %newVar36 <- 2
    %newVar36 *= %newVar35
    %newVar31 <- %newVar31
    %newVar31 += %newVar36
    %newVar31 <- %newVar31
    %newVar31 += 0
    %newVar30 <- %newVar31
    %newVar30 *= 8
    %newVar30 <- %newVar30
    %newVar30 += 32
    %newVar37 <- %arr
    %newVar37 += %newVar30
    mem %newVar37 0  <- 29
    %newVar39 <- 0
    %newVar40 <- %arr
    %newVar40 += 24
    %newVar41 <- mem %newVar40 0 
    %newVar43 <- 1
    %newVar42 <- %newVar41
    %newVar42 >>= 1
    %newVar43 <- %newVar43
    %newVar43 *= %newVar42
    %newVar44 <- 3
    %newVar44 *= %newVar43
    %newVar39 <- %newVar39
    %newVar39 += %newVar44
    %newVar39 <- %newVar39
    %newVar39 += 0
    %newVar38 <- %newVar39
    %newVar38 *= 8
    %newVar38 <- %newVar38
    %newVar38 += 32
    %newVar45 <- %arr
    %newVar45 += %newVar38
    mem %newVar45 0  <- 33
    %newVar47 <- 0
    %newVar48 <- %arr
    %newVar48 += 24
    %newVar49 <- mem %newVar48 0 
    %newVar51 <- 1
    %newVar50 <- %newVar49
    %newVar50 >>= 1
    %newVar51 <- %newVar51
    %newVar51 *= %newVar50
    %newVar52 <- 4
    %newVar52 *= %newVar51
    %newVar47 <- %newVar47
    %newVar47 += %newVar52
    %newVar47 <- %newVar47
    %newVar47 += 0
    %newVar46 <- %newVar47
    %newVar46 *= 8
    %newVar46 <- %newVar46
    %newVar46 += 32
    %newVar53 <- %arr
    %newVar53 += %newVar46
    mem %newVar53 0  <- 47
    %newVar55 <- 0
    %newVar56 <- %arr
    %newVar56 += 24
    %newVar57 <- mem %newVar56 0 
    %newVar59 <- 1
    %newVar58 <- %newVar57
    %newVar58 >>= 1
    %newVar59 <- %newVar59
    %newVar59 *= %newVar58
    %newVar60 <- 5
    %newVar60 *= %newVar59
    %newVar55 <- %newVar55
    %newVar55 += %newVar60
    %newVar55 <- %newVar55
    %newVar55 += 0
    %newVar54 <- %newVar55
    %newVar54 *= 8
    %newVar54 <- %newVar54
    %newVar54 += 32
    %newVar61 <- %arr
    %newVar61 += %newVar54
    mem %newVar61 0  <- 43
    %newVar63 <- 0
    %newVar64 <- %arr
    %newVar64 += 24
    %newVar65 <- mem %newVar64 0 
    %newVar67 <- 1
    %newVar66 <- %newVar65
    %newVar66 >>= 1
    %newVar67 <- %newVar67
    %newVar67 *= %newVar66
    %newVar68 <- 6
    %newVar68 *= %newVar67
    %newVar63 <- %newVar63
    %newVar63 += %newVar68
    %newVar63 <- %newVar63
    %newVar63 += 0
    %newVar62 <- %newVar63
    %newVar62 *= 8
    %newVar62 <- %newVar62
    %newVar62 += 32
    %newVar69 <- %arr
    %newVar69 += %newVar62
    mem %newVar69 0  <- 81
    %newVar71 <- 0
    %newVar72 <- %arr
    %newVar72 += 24
    %newVar73 <- mem %newVar72 0 
    %newVar75 <- 1
    %newVar74 <- %newVar73
    %newVar74 >>= 1
    %newVar75 <- %newVar75
    %newVar75 *= %newVar74
    %newVar76 <- 7
    %newVar76 *= %newVar75
    %newVar71 <- %newVar71
    %newVar71 += %newVar76
    %newVar71 <- %newVar71
    %newVar71 += 0
    %newVar70 <- %newVar71
    %newVar70 *= 8
    %newVar70 <- %newVar70
    %newVar70 += 32
    %newVar77 <- %arr
    %newVar77 += %newVar70
    mem %newVar77 0  <- 33
    %i <- 0
    %arrsize <- %arrsize
    %arrsize >>= 1
    goto :call_label2
  )
)
