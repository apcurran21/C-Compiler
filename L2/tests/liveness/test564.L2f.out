(
(in
(r12 r13 r14 r15 rbp rbx rdi)
(%num r12 r13 r14 r15 rbp rbx)
(%computes_remaining r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx rdi)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx rdi)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx rdi)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx rdi)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx rdi)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx rdi)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rax rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rax rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rax rbp rbx)
(%computes_remaining %computes_started %done %fib_i %fib_im1 r12 r13 r14 r15 rax rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 %past_base_case r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 %old_fib_i r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %old_fib_i r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
)

(out
(%num r12 r13 r14 r15 rbp rbx)
(%computes_remaining r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx rdi)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx rdi)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx rdi)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx rdi)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx rdi)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx rdi)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rax rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rax rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rax rbp rbx)
(%computes_remaining %computes_started %done %fib_i %fib_im1 r12 r13 r14 r15 rax rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rax rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 %past_base_case r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 %old_fib_i r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %old_fib_i r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(%computes_remaining %computes_started %fib_i %fib_im1 r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
()
)

)
